vancouver-imperial-college-london.csl

```diff
@@ -40,24 +44,29 @@
     <group delimiter=": ">
       <choose>
         <if type="chapter paper-conference entry-dictionary entry-encyclopedia" match="any">
-          <text term="in" text-case="capitalize-first"/>
+          <text term="in" text-case="capitalize-first" suffix=": "/>
         </if>
       </choose>
-      <names variable="editor">
-        <name sort-separator=" " initialize-with="" name-as-sort-order="all" delimiter=", " delimiter-precedes-last="always"/>
-        <label form="short" prefix=" (" suffix=")"/>
-      </names>
     </group>
+    <choose>
+      <if type="book" match="all" variable="author"/>
+      <else>
+        <names variable="editor">
+          <name sort-separator=" " initialize-with="" name-as-sort-order="all" delimiter=", " delimiter-precedes-last="always"/>
+          <label form="short" prefix=" (" suffix=")"/>
+        </names>
+      </else>
+    </choose>
   </macro>
   <macro name="publisher">
     <choose>
-      <if type="report" match="none">
+      <if type="report article article-journal" match="none">
         <group delimiter=": " suffix=";">
           <choose>
             <if type="thesis">
               <text variable="publisher-place" prefix="[" suffix="]"/>
             </if>
-            <else-if type="speech"/>
+            <else-if type="speech article-journal report" match="any"/>
```

Comment:
so just to confirm: reports would get neither publisher nor publisher-place? That seems unusual?

Reply:
Yes, that's been confirmed by our Liason librarians.: '...I think they have a point here, the publisher and place should be included for reports. However the print guide and web guide doesn’t include this information so for consistency I think we should stick with the status quo. When we look at referencing again next year we can decide whether to change this across the board....'